File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int>ans;
        int n=nums.size();
        vector<int>freqarr(n+1,0);
        for(int i=0;i<n;i++){
            freqarr[nums[i]]++;
        }
        for(int i=1;i<freqarr.size();i++){
            if(freqarr[i]==0){
                ans.push_back(i);
            }
        }
        return ans;
    }
};

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n=nums.length;
        List<Integer> ans = new ArrayList<>();
        int[] arr=new int[n+1];
       for(int i=0;i<nums.length;i++){
        arr[nums[i]]++;    
        }
        for(int i=1;i<=nums.length;i++){
            if(arr[i]==0){
                ans.add(i);
            }
        }
        return ans;
    }
}